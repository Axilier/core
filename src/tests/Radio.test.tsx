import React from 'react';
import { mount } from 'enzyme';
import { Radio } from '../index';

const onChange = jest.fn();
const radio = mount(<Radio options={['Test1', 'Test2']} onChange={onChange} />);

describe('when the radio is used in an application', () => {
    it('should render the contained radio correctly', () => {
        expect(radio).toMatchSnapshot();
    });
    describe('when I click the non selected button', () => {
        beforeAll(() =>
            radio
                .find('#Test2')
                .simulate('change', { target: { value: 'Test2' } }),
        );
        it('should change the visibly selected radio button', () => {
            expect(onChange).toHaveBeenCalledWith('Test2');
        });
    });
});
