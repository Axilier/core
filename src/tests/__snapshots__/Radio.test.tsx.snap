// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the radio is used in an application should render the contained radio correctly 1`] = `
<Radio
  onChange={[MockFunction]}
  options={
    Array [
      "Test1",
      "Test2",
    ]
  }
>
  <form
    className="contactRadio"
  >
    <label
      className="radioContainer"
      htmlFor="Test1"
      key="Test1"
    >
      Test1
      <input
        checked={true}
        id="Test1"
        name="radio"
        onChange={[Function]}
        type="radio"
        value="Test1"
      />
      <span
        className="radioCheckmark"
      />
    </label>
    <label
      className="radioContainer"
      htmlFor="Test2"
      key="Test2"
    >
      Test2
      <input
        checked={false}
        id="Test2"
        name="radio"
        onChange={[Function]}
        type="radio"
        value="Test2"
      />
      <span
        className="radioCheckmark"
      />
    </label>
  </form>
</Radio>
`;
